arch/x86/core/ia32/excstub.S

```diff
@@ -130,6 +130,7 @@ SECTION_FUNC(PINNED_TEXT, _exception_enter)
 #endif
 	/* ESP is pointing to the ESF at this point */
 
+#ifndef CONFIG_X86_MMU
```

Comment:
Isn't this a bit too heavy handed? It seems that interrupts should not be enabled only if the MMU is handling an exception. Not re-enabling interrupts at all if MMU is enabled would mean that if another kind of interrupt is being handled, interrupts won't be re-enabled as well. Can't this be done more specific to the MMU handling case? Am I missing something?
@dcpleung any thoughts here?

Reply:
I guess a better place to do this would be in the page fault handler where it disables interrupts until safe to re-enable again?

Reply:
no existing good way to know if current exception is page fault exception.
@dcpleung Your way still has a short window with interrupt enabled, which is still not safe.
And I doubt it's really necessary to enable interrupt during an exception handling. Fire an interrupt that could be any type during an exception handling, the ISR may wake up another thread with higher priority, and swap code after ISR will switch to that thread to run, leaving half-way exception handling.